.github/workflows/build-test-ship.yaml

```diff
@@ -183,34 +190,247 @@ jobs:
             ${{ env.ECR_REGISTRY }}/${{ env.ECR_REPOSITORY }}:${{ env.IMAGE_TAG }}
             ${{ env.ECR_REGISTRY }}/${{ env.ECR_REPOSITORY }}:latest
 
+  build-for-target-platforms:
+    name: Verify ${{ matrix.service }} images can be built on target platforms
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        service: ${{ fromJson(inputs.services) }}
+    if: github.event_name == 'pull_request' && !github.event.pull_request.draft
```

Comment:
I've restricted this job to non-draft PRs because it takes a long time and in my opinion doesn't provide much signal but is still worth running at least once before release.